modules/meshrenderinggl/include/modules/meshrenderinggl/datastructures/rasterization.h

```diff
@@ -31,20 +31,49 @@
 
 #include <modules/meshrenderinggl/meshrenderingglmoduledefine.h>
 #include <modules/opengl/shader/shaderutils.h>
+#include <modules/base/processors/transform.h>
 
 #include <vector>
 #include <unordered_map>
 #include <optional>
 
 namespace inviwo {
 
+namespace rasterization {
+
+struct SimpleSpatialEntity : public SpatialEntity<3> {
+    SimpleSpatialEntity() = default;
+    SimpleSpatialEntity(const SpatialEntity<3>& rhs) : SpatialEntity(rhs) {}
+    SimpleSpatialEntity(const SimpleSpatialEntity& rhs)
+        : SpatialEntity(rhs), replaceTransformationOnApply_(rhs.replaceTransformationOnApply_) {}
+    virtual SpatialEntity<3>* clone() const override { return new SimpleSpatialEntity(*this); }
+    SimpleSpatialEntity applyToSpatialEntity(const SpatialEntity<3>& data) const;
+
+    bool replaceTransformationOnApply_ = false;
+};
+
+struct TransformSettings : public CompositeProperty {
+    TransformSettings(const std::string& identifier = "transformSettings",
+                      const std::string& displayName = "Mesh Transforms",
+                      InvalidationLevel invalidate = InvalidationLevel::InvalidOutput);
+    TemplateOptionProperty<CoordinateSpace> space_;
+    BoolProperty replaceTransform_;
+    TransformListProperty transform_;
+
+    SimpleSpatialEntity getTransform() const;

```

Comment:
This is a bit counter-intuitive

Reply:
Which part exactly? This property mimics the interface of the transform processor. Since it does not have a SpatialEntity, it can not apply the transform directly, but rather return the transformation itself.
This might get clearer when SimpleSpatialEntity receives some docu.

Reply:
Depends on if one view: SpatialEntity = Transform

Reply:
Now named CompositeTransform.